File: Rubybooru.Core/Tag.cs
using System.Collections.Generic;

namespace Rubybooru.Core
{
    public class Tag
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public TagType Type { get; set; }
        public bool UserCreated { get; set; }
        
        public virtual List<ImageTag> Images { get; set; } = new List<ImageTag>();
    }
}

File: Rubybooru.Angular/src/app/entities/tag.ts
import { TagType } from './tag-type.enum';

export class Tag {
  id: number;
  name: string;
  type: TagType = 0;
  userCreated: boolean;
  count: number;

  constructor( id: number, name: string, userCreated: boolean = true ) {
    this.id = id;
    this.name = name;
  }
}
